features/FxA-45-device-registration-api/README.md

```diff
@@ -0,0 +1,386 @@
+https://api-accounts.dev.lcip.org/v1/account/devices
+```
+
+#### Response
+
+If the `sessionToken` field is `null`,
+it means the device is disconnected.
+
+```
+[
+  {
+    "id": "0f7aa00356e5416e82b3bef7bc409eef",
+    "sessionToken": "27cd4f4a4aa03d7d186a2ec81cbf19d5c8a604713362df9ee15c4f4a4aa03d7d",
+    "name": "My Phone",
+    "type": "mobile",
+    "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef",
+    "publicKey": "468601214f60f4828b6cd5d51d9d99d212e7c73657978955f0f5a5b7e2fa1370"
+  },
+  {
+    "id": "0f7aa00356e5416e82b3bef7bc409eef",
+    "sessionToken": null,
+    "name": "My Desktop",
+    "type": "desktop",
+    "callback": "https://updates.push.services.mozilla.com/update/d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c75",
+    "publicKey": "468601214f60f4828b6cd5d51d9d99d212e7c73657978955f0f5a5b7e2fa1370"
+  }
+]
+```
+
+### POST /v1/account/device/destroy
+
+Authenticated with session token or OAuth.
+
+#### Request
+
+```
+curl -v \
+-X POST \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/device/destroy \
+-d '{
+  "id": "0f7aa00356e5416e82b3bef7bc409eef"
+}'
+```
+
+#### Response
+
+```
+{}
+```
+
+## Details
+
+### How are shared devices handled?
+
+A shared device may or may not use the same device id for each user,
+there are tradeoffs for both.
+Either way, it should be difficult for the service to detect
+whether a device is shared and by whom.
+
+### Are shared devices bound to the account or the device?
+
+For example, Alice and Bob share a device.
+Alice registers the device with the name "Alice's Tablet".
+Later, Bob logs in and changes the name to "Bob's Tablet".
+The server still has her device name as "Alice's Tablet"
+but the device may have saved it locally as "Bob's Tablet".
+Should Alice's record be updated?
+When Alice logs in again, which name should she see?
+
+It can work either way, but we should decide on one or the other:
+
+  * Shared device with shared info;
+  * Or shared device with individual info?
+
+Danny's preference was for individual info.
+
+Ryan K commented:
+
+> It's not really clear how this should behave from device's individual perspective,
+> let alone when you put the server in the mix.
+> For e.g. a desktop firefox instance,
+> ISTM that "Bob logs in" implies that Alice gets logged out,
+> which could clear the locally-stored device name
+> and other FxA-related customizations.
+> For a dedicated Firefox device maybe it's not so simple.
```

Comment:
In our implementation, the device id is subsidiary to the account, so I don't think there's anything useful we can do with the notion of "shared devices".  If Bob logs in, the device now belongs to Bob as far as we're concerned.  We may be able to just remove this whole section.